src/components/createOnyxContext.tsx

```diff
@@ -86,7 +86,7 @@ export default <TOnyxKey extends OnyxKey>(onyxKeyName: TOnyxKey): CreateOnyxCont
         if (value === null) {
             throw new Error(`useOnyxContext must be used within a OnyxProvider [key: ${onyxKeyName}]`);
         }
-        return value;
+        return value as NonNullable<OnyxValue<TOnyxKey>>;
```

Comment:
We have a case where we create a context hook from `createOnyxContext` it might return undefined sometimes and we are not covering this case in our code, that is why app is crashing in specific scenario. More info on proposal: https://github.com/Expensify/App/issues/52633#issuecomment-2479496928